Replace dialog with dynamic card

bkbilly opened the pull request:
The popup window had issues when used with other cards, so I replaced it.
Now the card changes dynamically based on events which makes it simpler to use.

Keep in mind that this is a major change, so I would recommend to check out for any bugs, though on my tests it works great.

Updates:
 - Replace popup window with dynamic card (#47)
 - Hide video window and button if the user has disabled it
 - Input text for custom entities to dial any number (#69)

TECH7Fox commented:
Looks good, but the audioVisualizer isn't centered relative to the card.
![image](https://user-images.githubusercontent.com/32220029/166247771-9fa6da69-be52-40cf-b068-1bbfa88b91a2.png)

Also, I think some people still prefer the dialog because it gives more space for video and buttons, and works fine when you have a separate dashboard for it. So how about a option to set this inside a `<ha-dialog>` element?

And when I go back, the extension color becomes red, but there is no unregister in the log. Guess because it reloads the CSS as well?

bkbilly commented:
I just fixed the extension color. It was one of my recent changes that broke it.
The audioVisualizer seems a bit more complicated than what I thought, but I will try to fix it.

It's a good idea to have both the ha-dialog, and the dynamic card, but it can make things much more complicated. 
For now, one option is to write on README a section that explains how the previous versions work in respect with the newer.
This can give people the option to choose which one they like and then we can implement both.

bkbilly commented:
I've fixed the audioVisualizer and I've also removed the video components if the video is disabled.
As for the dialog window, I don't think it is necessary because you can create a new view and select the type Panel as you can see a preview bellow.

![image](https://user-images.githubusercontent.com/518494/166312194-964eda64-91ae-4d38-ab11-521e6b77cbce.png)

TECH7Fox commented:
> As for the dialog window, I don't think it is necessary because you can create a new view and select the type Panel as you can see a preview bellow.

Ah that's a good idea. But then we will need to increase the card size to span accross the entire panel, otherwise the video will be that small space. With the dialog it could stretch.

Will also do some more testing with video.

bkbilly commented:
I fixed a bug with the remoteAudio and I added the feature for calling any number from here #69.

TECH7Fox commented:
@bkbilly I made some CSS changes, could you test? Video should now fit correctly, the bottom will go over it, and in "panel mode" the popup should stretch over the entire page.

bkbilly commented:
I've tested it and everything seems to be working great!
I believe that this can be the next release.

I have some ideas for the future releases, though they are not that important.
I am just gonna mention them here:
 - Option to choose between `wss` and `ws`.
 - Extensions person should be optional so that other users can register if they are not in the list.
 - Option to hide other extensions.
 - For custom edit entities, it should call the number by pressing enter key.

TECH7Fox commented:
Thanks, will merge then.

>  Option to choose between wss and ws.

Option to use `ws` wouldn't work, because browsers don't accept that anymore for security reasons.

> Extensions person should be optional so that other users can register if they are not in the list.

We could make a `default` option for a extension, so when the person isn't found, it will use that one?

> Option to hide other extensions.

Another extension option, that would always hide that extension?

TECH7Fox commented:
Made the new version: https://github.com/TECH7Fox/sip-hass-card/releases/tag/v2.2.0

Thanks @bkbilly! 🙂
